torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -608,6 +623,9 @@ XLAGraphExecutor::ExecuteComputationWithBarrier(
       if (auto xla_tensor_ptr = bridge::TryGetXlaTensor(ivalue.toTensor())) {
         dataptr = xla_tensor_ptr->GetXlaData();
       } else {
+        XLA_CHECK(device.type() != (int8_t)XlaDeviceType::SPMD)
```

Comment:
What's this XLA_CHECK for?

Reply:
hmm, not sure, I copy this from @yeounoh 's diff. @yeounoh any idea?

Reply:
It's not needed, but more for a sanity check I probably added to ensure that this doesn't happen. Basically, we want to make sure that the SPMD device type is always on the backend (device data).